File: OC/IsCompleteBlock/IsCompleteBlock/ViewController.m
//
//  ViewController.m
//  IsCompleteBlock
//
//  Created by hrt03 on 2019/12/18.
//  Copyright © 2019 dengchaojie. All rights reserved.
//

#import "ViewController.h"
#import "JudgeCardData.h"

@interface ViewController ()
@property (nonatomic, strong) JudgeCardData *judge;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    JudgeCardData *judge = JudgeCardData.new;
    _judge = judge;
    [_judge judgeCardData:^(BOOL isValid, NSString * _Nonnull msg) {
        if (isValid) {
            
        } else {
            NSLog(@"%@", msg);
        }
    }];
    
    
}


@end

File: Combine/DemoCombine/DemoCombine/ViewController.swift
//
//  ViewController.swift
//  DemoCombine
//
//  Created by hrt03 on 2019/9/24.
//  Copyright © 2019 dengchaojie. All rights reserved.
//

import UIKit
import Combine

class ViewController: UIViewController {
        
    let signInPublisher = NotificationCenter.Publisher(center: .default, name: HrtConstraint.nameNote).map {
        return ($0 as? SignInInfo)?.userName ?? "unknown"
    }
    
    let nameLabel = UILabel.init(frame: CGRect.init(x: 100, y: 100, width: 100, height: 40))
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        view.addSubview(nameLabel)
        self.registerSignInHandle()
//        let info = SignInInfo.init(userName: "dcj", password: "123")
    }
    
    func registerSignInHandle() {
        let signInSubscriber = Subscribers.Assign.init(object: self.nameLabel, keyPath: \.text)
//        signInPublisher.subscribe(signInSubscriber)
        
    }

}

